### PR TITLE
gitlab: 16.3.1 -> 16.3.2

### PR DESCRIPTION
**Pull request by yayayayaka**
## Description of changes
https://gitlab.com/gitlab-org/gitlab/-/blob/v16.3.2-ee/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by kip93 — APPROVED** [review]

File: pkgs/applications/version-management/gitlab/gitaly/default.nix [hunk 1]
```diff
@@ -24,7 +24,7 @@ let
       owner = "gitlab-org";
       repo = "gitaly";
       rev = "v${version}";
-      sha256 = "sha256-Tc7JsGuTF+JK5SLmaQwVbmdSy/Wvm3VNoZqHTyZFd6k=";
+      sha256 = "sha256-Q2wXoc4tLJvc0LwSct3cCswOqhUJSVVhoG/uFXeVOB4=";
```
Comment (Janik-Haag) [inline pkgs/applications/version-management/gitlab/gitaly/default.nix:27]:
```suggestion
      hash = "sha256-Q2wXoc4tLJvc0LwSct3cCswOqhUJSVVhoG/uFXeVOB4=";
```

File: pkgs/applications/version-management/gitlab/gitlab-pages/default.nix [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
   pname = "gitlab-pages";
-  version = "16.3.1";
+  version = "16.3.2";
 
   src = fetchFromGitLab {
     owner = "gitlab-org";
     repo = "gitlab-pages";
     rev = "v${version}";
-    sha256 = "sha256-gUUmUVBk6Ox4oKOef8RDqmW2Hs2wQL/lpbIEbwbzgHY=";
+    sha256 = "sha256-1e3s+RLuNilNtsKzfKrsbAD0dzA87LbLH/c43CKh5EU=";
```
Comment (Janik-Haag) [inline pkgs/applications/version-management/gitlab/gitlab-pages/default.nix:11]:
```suggestion
    hash = "sha256-1e3s+RLuNilNtsKzfKrsbAD0dzA87LbLH/c43CKh5EU=";
```

**Review by Janik-Haag — COMMENTED** [review]
just a small nitpick, the rest looks good to me

File: pkgs/applications/version-management/gitlab/gitaly/default.nix (hunk 1 above)
Comment (yayayayaka) [reply pkgs/applications/version-management/gitlab/gitaly/default.nix]: These are automatically generated through the update script.

Comment (Janik-Haag) [reply pkgs/applications/version-management/gitlab/gitaly/default.nix]: nix-update only exchanges the value, but works with both `sha256` and `hash` and `hash` is the new one you are supposed to use now.

**Pull request merged**